src/Workspaces/Core/Portable/Editing/ImportAdderService.cs

```diff
@@ -92,8 +89,13 @@ public enum Strategy
 
             return document.WithSyntaxRoot(root);
 
-            bool IsInSpan(SyntaxNode node) =>
-                spansTree.HasIntervalThatOverlapsWith(node.FullSpan.Start, node.FullSpan.Length);
+            bool IsInSpan(SyntaxNode node)
+            {
+                // Create a simple interval tree for simplification spans.
+                var spansTree = new SimpleIntervalTree<TextSpan>(TextSpanIntervalIntrospector.Instance, spans);
```

Comment:
😟 Did this just go from O(1) to O(n)?

Reply:
Good catch. I'll revert

---
In reply to: [364502110](https://github.com/dotnet/roslyn/pull/40755#discussion_r364502110) [](ancestors = 364502110)